ibis/backends/base_sql/__init__.py

```diff
@@ -5,10 +5,21 @@
 from the SQL-based backends.
 """
 import datetime
+import itertools
```

Comment:
rather than defining everything here, move to modules and import *only* what is needed into the namespace

Reply:
Done. I left datetime global because it was used by two functions, but let me know if you would prefer moving it into them as well.